internal/server/statvar/statvar_group.go

```diff
@@ -114,6 +116,51 @@ func filterSVG(in *pb.StatVarGroups, statVars []string) *pb.StatVarGroups {
 	return result
 }
 
+// adjustDescendentStatVarCount returns the DescendentStatVarCount for node curID.
+// Mutates DescendentStatVarCount for all nodes in id2Node starting at customSvgRoot.
+func adjustDescendentStatVarCount(id2Node map[string]*pb.StatVarGroupNode, curID string) int32 {
+	curNode, ok := id2Node[curID]
+	if !ok {
+		return 0
+	}
+	curNode.DescendentStatVarCount = int32(len(curNode.ChildStatVars))
+	for _, childSVG := range curNode.GetChildStatVarGroups() {
+		// Child SVG protos have its own DescendentStatVarCount that must be set.
+		childSVG.DescendentStatVarCount = adjustDescendentStatVarCount(id2Node, childSVG.GetId())
+		curNode.DescendentStatVarCount += childSVG.DescendentStatVarCount
+	}
+	return curNode.GetDescendentStatVarCount()
+}
+
+// mergeSVGNodes merges node2's svg and svs into node1.
+// Merge here refers to set operation.
+// Warning: node1's DescendentStatVarCount is unchanged.
+// Caller is responsible for calling fixCustomRootDescendentStatVarCount
+// at the end of all merges.
+func mergeSVGNodes(node1, node2 *pb.StatVarGroupNode) {
```

Comment:
I wonder if this should be recursive?

Reply:
This is not recursive because the calling function happens while the id -> [child node] map is being updated. This is caller per update for the merging of two nodes. 

As spoken offline, for main DC, we can take the max. On second though, I think we should take the sum instead of max? ex: node1[count=2] and node2[count=3], the actual count is between 3 and 5.